CMakePresets.json

```diff
@@ -0,0 +1,216 @@
+{
+  "version": 3,
+  "cmakeMinimumRequired": {
+    "major": 3,
+    "minor": 21,
+    "patch": 0
+  },
+  "configurePresets": [
+    {
+      "name": "nix-base",
+      "hidden": true,
+      "generator": "Ninja",
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER_LAUNCHER": "ccache",
+        "CMAKE_FIND_FRAMEWORK": "LAST",
+        "CMAKE_FIND_PACKAGE_PREFER_CONFIG": "ON",
+        "CMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY": "OFF",
+        "CMAKE_FIND_USE_PACKAGE_REGISTRY": "OFF",
+        "CMAKE_OSX_SYSROOT": "",
+        "VAST_ENABLE_DEVELOPER_MODE": "ON",
+        "VAST_ENABLE_LSVAST": "ON",
+        "VAST_ENABLE_MDX_REGENERATE": "ON",
```

Comment:
It's `s/MDX_REGENERATE/VAST_REGENERATE`. I also don't think that tool should be enabled by default. We don't have any tests for it and it is highly dangerous to use, so we should never be shipping it with any VAST packages.

Reply:
These presets are for development, not distribution.

Reply:
It is `MDX_REGENERATE` btw.

Reply:
https://github.com/tenzir/vast/pull/2260 changes that, so please at least leave a comment in there to change the preset accordingly.

I don't see how this would only be for development and not for distribution. If I build the preset `nix-static-release` I get this, and that's exactly what we distribute.

Reply:
> I don't see how this would only be for development and not for distribution. If I build the preset nix-static-release I get this, and that's exactly what we distribute.

The package we distribute is not built with these presets but via https://github.com/tenzir/vast/blob/master/nix/vast/default.nix. If that file starts making use of the preset we can still add overrides with the `cmakeFlags` variable.

Reply:
Unresolving this because the other PR is now merged, so we have to `s/MDX_REGENERATE/VAST_REGENERATE`